src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs

```diff
@@ -237,21 +237,39 @@ internal static OtlpCommon.KeyValue ToOtlpAttribute(this KeyValuePair<string, ob
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static OtlpTrace.Status ToOtlpStatus(ref TagEnumerationState otlpTags)
+        private static OtlpTrace.Status ToOtlpStatus(this Activity activity, ref TagEnumerationState otlpTags)
         {
-            var status = StatusHelper.GetStatusCodeForTagValue(otlpTags.StatusCode);
-
-            if (!status.HasValue)
+            if (activity.Status == ActivityStatusCode.Unset && StatusHelper.GetStatusCodeForTagValue(otlpTags.StatusCode) == null)
             {
                 return null;
             }
 
-            var otlpStatus = new OtlpTrace.Status
+            OtlpTrace.Status.Types.StatusCode otlpActivityStatusCode = OtlpTrace.Status.Types.StatusCode.Unset;
+            string otlpStatusDescription = null;
+            if (activity.Status != ActivityStatusCode.Unset)
             {
                 // The numerical values of the two enumerations match, a simple cast is enough.
-                Code = (OtlpTrace.Status.Types.StatusCode)(int)status,
-            };
+                otlpActivityStatusCode = (OtlpTrace.Status.Types.StatusCode)(int)activity.Status;
+                if (activity.Status == ActivityStatusCode.Error && !string.IsNullOrEmpty(activity.StatusDescription))
+                {
+                    otlpStatusDescription = activity.StatusDescription;
+                }
+            }
+            else
+            {
+                var statusCode = StatusHelper.GetStatusCodeForTagValue(otlpTags.StatusCode);
+                if (statusCode != StatusCode.Unset)
+                {
+                    // The numerical values of the two enumerations match, a simple cast is enough.
+                    otlpActivityStatusCode = (OtlpTrace.Status.Types.StatusCode)(int)statusCode;
+                    if (statusCode == StatusCode.Error && !string.IsNullOrEmpty(otlpTags.StatusDescription))
```

Comment:
currently, there is no check for statusCode == Error or not. This is adding that check to existing path, which is not required.

Reply:
I guess you are referring to similar issue as this?
https://github.com/open-telemetry/opentelemetry-dotnet/pull/3100#discussion_r836918405

Reply:
Yes! Thanks for pointing to the previous discussion.
(Please update PR description to include this, or ideally this is a separate bug fix , and can be fixed in separate PRs.)
(Not blocker btw)